hail/python/hailctl/dataproc/start.py

```diff
@@ -172,6 +173,9 @@ def main(args, pass_through_args):
     if args.bucket:
         conf.flags['bucket'] = args.bucket
 
+    label = safe_call('gcloud', 'config', 'get-value', 'account')
+    conf.flags['labels'] = 'creator=' + label.replace('.', '-dot-').replace('@', '-at-')  # no special chars
```

Comment:
This seems insufficient to be reliable.  From: https://cloud.google.com/resource-manager/docs/creating-managing-labels

 > Values can be empty, and have a maximum length of 63 characters.  ... Keys and values can contain only lowercase letters, numeric characters, underscores, and dashes.  

Emails can have other characters (upper case, +, etc.)

Reply:
yeah, you're totally right. I'll write something that checks against a regex and defaults to not labeling and printing an error.

Reply:
That means if the user has a weird email they can't use gcloud?  Seems rude.

I was going to suggest just munging it: lowercase-ify, replace expected characters readably like you did, change other invalid characters to just _, and then truncate.  As long as this preserves enough information that Sam can figure out who ran it, we're golden.

Reply:
Err, can't use hailctl.

Reply:
but yeah, your solution is better